fixed review to only show 1

asgermunk opened the pull request.

asgermunk committed:
Add ListAllCriteria method and make Grade nullable

Added ListAllCriteria method to IConferenceRepo interface to list all criteria for a given conference. Implemented a placeholder for this method in LocalConferenceRepo. Changed Grade property in Criteria class to nullable int for more flexible data handling.

asgermunk committed:
Add review update methods and new FinalReviewComponent

- Re-added `SolutionProperties` in `conferencePlanner.sln`.
- Added `UpdateReview` method to `IAbstractRepo` and `IAbstractService`.
- Implemented `UpdateReview` in `LocalAbstractRepo` and `LocalStorageAbstractService`.
- Simplified `ReviewComponent.razor` to show only the first review.
- Renamed method `UpdateAbstract` to `UpdateReview` in `ReviewComponent.razor`.
- Removed unused CSS styles in `ReviewComponent.razor`.
- Added `FinalReviewComponent.razor` to display average review grades.

asgermunk committed:
Merge branch 'development' into final-review-component

asgermunk committed:
Add final review display and comments section

Updated `FinalReviewComponent.razor` to show the first review and a new comments section if reviews are available for the selected abstract. Added corresponding CSS styles for the comments section and comment cards. Modified `ReviewPage.razor` to include `FinalReviewComponent` in the review column to display final review details.

asgermunk committed:
Merge branch 'development' into final-review-component

asgermunk committed:
now updates properly. and is finished.

asgermunk committed:
Add null and count check for SelectedAbstract.Reviews

Added a check to ensure that `SelectedAbstract?.Reviews` is not null and contains at least one review before proceeding with the update. If the check fails, the method returns early without performing any update. Moved the `isSaving = true;` line to execute after the new check and removed the unnecessary line break before setting `isSaving` to true.

asgermunk committed:
fixed things

Mikkelkorsaa commented on conferencePlanner/conferencePlannerApi/Repositories/Interfaces/IAbstractRepo.cs:
```diff
@@ -25,5 +25,7 @@ public interface IAbstractRepo
                 Task<bool> DeleteAsync(int id);
-        }
+
+                Task<Abstract?> UpdateReview(int abstractId, Review review);
```
Fix nullable

Mikkelkorsaa commented on conferencePlanner/conferencePlannerApp/Components/FinalReviewComponent.razor:
```diff
@@ -0,0 +1,157 @@
+        var averageCriteriaGrades = new List<AverageCriteria>();
+
+        if (SelectedAbstract?.Reviews != null)
+        {
```
Fix nullable

Mikkelkorsaa commented on conferencePlanner/conferencePlannerApp/Components/ReviewComponent.razor:
```diff
@@ -4,39 +4,35 @@
     <div class="abstract-review">
         @if (SelectedAbstract?.Reviews != null && SelectedAbstract.Reviews.Any())
         {
```
Fix nullable

Mikkelkorsaa commented on conferencePlanner/conferencePlannerApp/Components/ReviewComponent.razor:
```diff
@@ -4,39 +4,35 @@
+                    }
+
+                    <div class="form-group">
+                        <label class="comment-label">Comment</label>
```
"Comment" = "Kommentar"

Mikkelkorsaa commented on conferencePlanner/conferencePlannerCore/Models/Criteria.cs:
```diff
@@ -9,6 +9,6 @@ namespace conferencePlannerCore.Models
     {
         public string Name { get; set; } = string.Empty;
-        public int Grade { get; set; }
+        public int? Grade { get; set; }
```
Er det rigtigt??

asgermunk replied:
ja

Mikkelkorsaa reviewed (changes requested):
Pls fix

asgermunk committed:
Update method signatures and UI labels

Modified `UpdateReview` in `IAbstractRepo.cs` to return non-nullable `Task<Abstract>` and added comments for clarity. Updated `FinalReviewComponent.razor` and `ReviewComponent.razor` to change "Comments" to "Kommentar" and replaced null-conditional operators with null-forgiving operators for `SelectedAbstract.Reviews` property checks.
